7.technical_analysis/scripts/0.preprocess_temporary_dirs.py

```diff
@@ -0,0 +1,333 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import os
+import pathlib
+import sys
+
+import matplotlib.pyplot as plt
+
+# Import dependencies
+import numpy as np
+import pandas as pd
+import seaborn as sns
+import skimage
+import tifffile
+from matplotlib.colors import BoundaryNorm, ListedColormap
+from skimage import io
+
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+sys.path.append(str(root_dir / "utils"))
+from arg_parsing_utils import check_for_missing_args, parse_args
+from file_reading import read_zstack_image
+from notebook_init_utils import bandicoot_check, init_notebook
+
+root_dir, in_notebook = init_notebook()
+
+if in_notebook:
+    from tqdm.notebook import tqdm
+else:
+    from tqdm import tqdm
+
+image_base_dir = bandicoot_check(
+    pathlib.Path(os.path.expanduser("~/mnt/bandicoot")).resolve(), root_dir
+)
+
+
+# In[2]:
+
+
+def convert_indexed_mask_to_binary_mask(indexed_mask: np.ndarray) -> np.ndarray:
+    """
+    Convert an indexed mask to a binary mask.
+
+    Parameters
+    ----------
+    indexed_mask : np.ndarray
+        An indexed mask where 0 represents the background and any positive integer represents a different object.
+
+    Returns
+    -------
+    np.ndarray
+        A binary mask where True represents the foreground (objects) and False represents the background.
+    """
+    binary_mask = np.zeros_like(indexed_mask, dtype=bool)
+    binary_mask[indexed_mask > 0] = True
+    return binary_mask
+
+
+def extract_IOU(mask1: np.ndarray, mask2: np.ndarray) -> float:
+    """
+    Calculate the Intersection over Union (IoU) between two binary masks.
+
+    Parameters
+    ----------
+    mask1 : np.ndarray
+        A binary mask where True represents the foreground (objects) and False represents the background.
+    mask2 : np.ndarray
+        A binary mask where True represents the foreground (objects) and False represents the background.
+
+    Returns
+    -------
+    float
+        The Intersection over Union (IoU) score between the two masks.
+    """
+    intersection = np.logical_and(mask1, mask2)
+    union = np.logical_or(mask1, mask2)
+    iou_score = np.sum(intersection) / np.sum(union)
```

Comment:
Always good to prevent division by zero, you can also define `eps=1e-12` or your favorite small number and use that
```suggestion
    iou_score = np.sum(intersection) / (np.sum(union) + 1e-12)
```